salt/tests/unit/modules/files/test_metalk8s_volumes.yaml

```diff
@@ -260,13 +415,45 @@ create:
     ftruncate: False
     raise_msg: "cannot create sparse file at .*: An error has occurred"
 
+  ## SPARSE block volume
+  # create a simple sparse volume
+  - name: my-sparse-block-volume
+    pillar_volumes: *volumes_details
+
+  # unable to truncate the sparse file
+  - name: my-sparse-block-volume
+    pillar_volumes: *volumes_details
+    ftruncate: False
+    raise_msg: "cannot create sparse file at .*: An error has occurred"
+
   ## RAW BLOCK DEVICE volume
   # create on raw block device (nothing to create)
   - name: my-raw-block-device-volume
     pillar_volumes: *volumes_details
     # We do not need to create block device
     raise_msg: block device /dev/sda1 does not exists
 
+  ## RAW BLOCK DEVICE block disk volume
+  # create on raw block device (nothing to create)
+  - name: my-raw-block-device-block-disk-volume
+    pillar_volumes: *volumes_details
+    # We do not need to create block device
+    raise_msg: block device /dev/sdb does not exists
```

Comment:
Not related to the test, but something from the original module (I just realized by reading your test cases :slightly_smiling_face:), this error message is strange, no? Shouldn't it say something like "can not create raw block devices"?

Reply:
Hmm maybe, the thing is just that the user should not call this function for raw block device.

But indeed the message is not really clear about it, 
it's maybe more a question for @slaperche-scality

Reply:
The reason behind `'block device {} does not exists'` is because if we ends up here, it means that `exists()` returned `False` (hence the `does not exists` error). Which can be a bit surprising when reading the code, but gives a good error message from Salt.

`"can not create raw block devices"` isn't really good, because we don't even try to create it.

Reply:
Ah right, I get it. So the test for `create()` on block device should try with an existing one (which would succeed) and a missing one (which would fail, because we can't `create()` a block device).
OK so that could have been made a bit clearer in the test cases

Reply:
Oh ok got it

> So the test for create() on block device should try with an existing one (which would succeed) and a missing one (which would fail, because we can't create() a block device).

No, it's not really this, calling `create` on raw block device will never succeed, it will always raise this error, because, I think, we do not call the `create` function when `exists` return True (which is always the case for raw block device) but for test purpose we should maybe define an "un-existing" volume so that the test case is easier to understand

Reply:
Yeah, you're not suppose to call `create` on a raw block device (sure you can, but as expected it will always fail because you can't create a raw block device out of thin air).
In the Salt state, it's guarded by a call to `exists`.